common/djangoapps/student/tasks.py

```diff
@@ -0,0 +1,48 @@
+"""
+This file contains celery tasks for sending email
+"""
+from django.conf import settings
+from django.core import mail
+
+from celery.task import task  # pylint: disable=no-name-in-module, import-error
+from celery.utils.log import get_task_logger  # pylint: disable=no-name-in-module, import-error
+from celery.exceptions import MaxRetriesExceededError
+log = get_task_logger(__name__)
+
+
+@task(bind=True)
+def send_activation_email(self, user, subject, message, from_address):
+    """
+    Sending an activation email to the users.
+    """
+    max_retries = settings.RETRY_ACTIVATION_EMAIL_MAX_ATTEMPTS
+    retries = self.request.retries + 1
+    dest_addr = user.email
+    try:
+        if settings.FEATURES.get('REROUTE_ACTIVATION_EMAIL'):
+            dest_addr = settings.FEATURES['REROUTE_ACTIVATION_EMAIL']
+            message = ("Activation for %s (%s): %s\n" % (user, user.email, user.profile.name) +
+                       '-' * 80 + '\n\n' + message)
+            mail.send_mail(subject, message, from_address, [dest_addr], fail_silently=False)
+        else:
+            user.email_user(subject, message, from_address)
+        # Log that the Activation Email has been sent to user without an exception
+        log.info("Activataion Email has been sent to User {user_email}".format(
+            user_email=dest_addr
+        ))
+    except Exception as exc:  # pylint: disable=broad-except
+        log.info('Retrying sending email to user {dest_addr}, attempt # {attempt} of {max_attempts}'. format(
+            dest_addr=dest_addr,
+            attempt=retries,
+            max_attempts=max_retries + 1
+        ))
+        try:
+            self.retry(countdown=settings.RETRY_ACTIVATION_EMAIL_TIMEOUT, max_retries=max_retries)
+        except MaxRetriesExceededError:
```

Comment:
@adampalay yeah, you are right. But when I am trying to create an account on my local environment, every time the email is sent. In addition to this, there are more than one exceptions that may be raised causing `send_email` to fail. 

If we handle only this exception `"NoAuthHandlerFound"` which is mentioned in this [comment](https://openedx.atlassian.net/browse/ECOM-5281?focusedCommentId=211754&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-211754). then no other exception other than this will be caught.